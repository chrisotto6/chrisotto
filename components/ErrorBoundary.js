import * as Sentry from '@sentry/nextjs'
import * as React from 'react'

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props)
    this.state = { hasError: false }
  }
  static getDerivedStateFromError(error) {
    return { hasError: true }
  }
  componentDidCatch(error, errorInfo) {
    // You can use your own error logging service here
    Sentry.captureException(errorInfo)
  }
  render() {
    // Check if the error is thrown
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return (
        <div>
          <h2>Oops, there is an error!</h2>
          <button type="button" onClick={() => this.setState({ hasError: false })}>
            Try again?
          </button>
        </div>
      )
    }
    return this.props.children
  }
}

export default ErrorBoundary
