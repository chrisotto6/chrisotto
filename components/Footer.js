import Link from 'next/link'
import siteMetadata from '@/data/siteMetadata'
import { FaEnvelope, FaGithub, FaLinkedin, FaTwitter } from 'react-icons/fa'

export default function Footer() {
  return (
    <footer>
      <div className="mt-16 flex flex-col items-center">
        <div className="mb-5 flex space-x-2 text-sm text-gray-500 dark:text-gray-400">
          <Link href="/">{siteMetadata.title}</Link>
          <div>{` • `}</div>
          <div>{`© ${new Date().getFullYear()}`}</div>
        </div>
        <div className="mb-3 flex space-x-8">
          <Link href={`mailto:${siteMetadata.email}`}>
            <FaEnvelope size={30} />
          </Link>
          <Link href={siteMetadata.github}>
            <FaGithub size={30} />
          </Link>
          <Link href={siteMetadata.linkedin}>
            <FaLinkedin size={30} />
          </Link>
          <Link href={siteMetadata.twitter}>
            <FaTwitter size={30} />
          </Link>
        </div>
      </div>
    </footer>
  )
}
