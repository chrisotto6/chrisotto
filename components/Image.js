import NextImage from 'next/image'

// eslint-disable-next-line jsx-a11y/alt-text
const Image = ({ ...rest }) => (
  <NextImage
    {...rest}
    style={{
      maxWidth: '100%',
      height: 'auto',
    }}
    sizes="100vw"
  />
)

export default Image
