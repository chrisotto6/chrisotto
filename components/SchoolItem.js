import kebabCase from '@/lib/utils/kebabCase'

const SchoolItem = ({ schoolName, degree, gpa, awards }) => {
  return (
    <div data-testid={`school-${kebabCase(schoolName)}`}>
      <h3>{schoolName}</h3>
      <h4>{degree}</h4>
      <h5>{`${gpa} - ${awards}`}</h5>
    </div>
  )
}

export default SchoolItem
