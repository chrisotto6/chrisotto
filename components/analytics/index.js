import GA from './GoogleAnalytics'
import siteMetadata from '@/data/siteMetadata'

const isProduction = process.env.NODE_ENV === 'production'

const Analytics = () => {
  return <>{isProduction && siteMetadata.analytics.googleAnalyticsId && <GA />}</>
}

export default Analytics
