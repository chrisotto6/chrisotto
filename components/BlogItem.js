import Image from './Image'
import Link from 'next/link'
import Tag from '@/components/Tag'
import formatDate from '@/lib/utils/formatDate'

const BlogItem = ({ slug, date, title, summary, tags, image, readingTime }) => (
  <article
    // prettier-ignore
    className="display:block w-full gap-6 space-y-2 rounded-xl xl:grid xl:grid-cols-2 xl:items-start dark:hover:bg-sky-700/40 hover:bg-gray-100"
    data-testid="list-blog-item"
  >
    <Link href={`/blog/${slug}`} className="text-gray-900 dark:text-gray-100">
      <Image className="w-full rounded-xl" alt={title} src={image} width={450} height={250} />
    </Link>
    <div className="space-y-6">
      <div className="space-y-5 xl:col-span-3">
        <Link href={`/blog/${slug}`} className="text-gray-900 dark:text-gray-100">
          <div>
            <h2 className="text-2xl font-bold leading-8 tracking-tight text-gray-900 dark:text-gray-100">
              {title}
            </h2>
            <dl>
              <dt className="sr-only">Published on</dt>
              <dd className="inline-block text-base font-medium leading-6 text-gray-500 dark:text-gray-400">
                <time dateTime={date}>{formatDate(date)}</time>
                <p>{readingTime.text}</p>
              </dd>
            </dl>
          </div>
          <div className="prose max-w-none text-gray-500 dark:text-gray-400">{summary}</div>
        </Link>
        <div className="flex flex-wrap" data-testid="tag-container">
          {tags.map((tag) => (
            <Tag key={tag} text={tag} />
          ))}
        </div>
      </div>
    </div>
  </article>
)

export default BlogItem
