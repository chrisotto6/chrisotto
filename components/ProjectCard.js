import Image from './Image'
import Link from 'next/link'
import { FaGithub, FaExternalLinkSquareAlt } from 'react-icons/fa'

const ProjectCard = ({ title, description, img, githubUrl, projectUrl, tech, dataTestid }) => (
  <div
    className="max-w-xs overflow-hidden rounded-xl shadow-lg dark:bg-gray-600 dark:shadow-cyan-600/50"
    data-testid={dataTestid}
  >
    <Image className="w-full" alt={title} src={img} width={544} height={306} />
    <div className="px-6 py-4">
      <div className="mb-2 text-xl font-bold">{title}</div>
      <p className="text-base text-gray-500 dark:text-gray-100">{description}</p>
    </div>
    <div className="flex px-6 pt-3 pb-2">
      {tech.map((t) => (
        <span
          key={t}
          className="mr-2 mb-2 inline-block rounded-full bg-gray-200 px-3 py-1 text-sm font-semibold text-gray-500 dark:bg-cyan-600 dark:text-gray-100"
        >
          {t}
        </span>
      ))}
    </div>
    <div className="mb-3 flex space-x-4 px-6 pt-3 pb-3">
      <Link href={githubUrl}>
        <FaGithub size={30} />
      </Link>
      <Link href={projectUrl}>
        <FaExternalLinkSquareAlt size={30} />
      </Link>
    </div>
  </div>
)

export default ProjectCard
