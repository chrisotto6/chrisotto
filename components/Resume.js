import schoolData from '@/data/schoolData'
import volunteerData from '@/data/volunteerData'
import workData from '@/data/workData'
import SchoolItem from '@/components/SchoolItem'
import VolunteerItem from '@/components/VolunteerItem'
import WorkItem from '@/components/WorkItem'
import { title } from '@/data/siteMetadata'

const Resume = () => {
  return (
    <article>
      <h2>Experience</h2>
      {workData.map((job) => (
        <WorkItem key={job} company={job.company} role={job.role} />
      ))}
      <h2>Education</h2>
      {schoolData.map((school) => (
        <SchoolItem
          key={school.schoolName}
          degree={school.degree}
          gpa={school.gpa}
          awards={school.awards}
        />
      ))}
      <h2>Volunteering</h2>
      {volunteerData.map((vol) => [
        <VolunteerItem
          key={vol.company}
          company={vol.company}
          title={vol.title}
          summary={vol.summary}
          yearStart={vol.yearStart}
          yearEnd={vol.yearEnd}
        />,
      ])}
    </article>
  )
}

export default Resume
