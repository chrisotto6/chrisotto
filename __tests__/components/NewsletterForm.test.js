import { render, fireEvent } from '@testing-library/react'
import NewsletterForm from '@/components/NewsletterForm'
import { BlogNewsletterForm } from '@/components/NewsletterForm'
import { enableFetchMocks } from 'jest-fetch-mock'

enableFetchMocks()

it('renders NewsletterForm unchanged', () => {
  const { container } = render(<NewsletterForm title="test" />)
  expect(container).toMatchSnapshot()
})

it('renders BlogNewsletterForm unchanged', () => {
  const { container } = render(<BlogNewsletterForm title="test" />)
  expect(container).toMatchSnapshot()
})

//it('submits test email signup', () => {
//  fetch.mockResponseOnce(() => Promise.resolve({message: 'success'}))
//  const { getByTestId } = render(<NewsletterForm title='test' />)
//  fireEvent.submit(getByTestId('email-form'))
//  expect(subscribe).toHaveBeenCalled()
//})
