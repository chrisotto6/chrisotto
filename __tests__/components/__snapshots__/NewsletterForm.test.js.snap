// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BlogNewsletterForm unchanged 1`] = `
<div>
  <div
    class="flex items-center justify-center"
  >
    <div
      class="bg-gray-100 p-6 dark:bg-gray-800 sm:px-14 sm:py-8"
    >
      <div>
        <div
          class="pb-1 text-lg font-semibold text-gray-800 dark:text-gray-100"
        >
          test
        </div>
        <form
          class="flex flex-col sm:flex-row"
          data-testid="email-form"
        >
          <div>
            <label
              class="sr-only"
              for="email-input"
            >
              Email address
            </label>
            <input
              autocomplete="email"
              class="w-72 rounded-md px-4 focus:border-transparent focus:outline-none focus:ring-2 focus:ring-primary-600 dark:bg-black"
              id="email-input"
              name="email"
              placeholder="Enter your email"
              required=""
              type="email"
            />
          </div>
          <div
            class="mt-2 flex w-full rounded-md shadow-sm sm:mt-0 sm:ml-3"
          >
            <button
              class="w-full rounded-md bg-primary-500 py-2 px-4 font-medium text-white sm:py-0 hover:bg-primary-700 dark:hover:bg-primary-400 focus:outline-none focus:ring-2 focus:ring-primary-600 focus:ring-offset-2 dark:ring-offset-black"
              type="submit"
            >
              Sign up
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders NewsletterForm unchanged 1`] = `
<div>
  <div>
    <div
      class="pb-1 text-lg font-semibold text-gray-800 dark:text-gray-100"
    >
      test
    </div>
    <form
      class="flex flex-col sm:flex-row"
      data-testid="email-form"
    >
      <div>
        <label
          class="sr-only"
          for="email-input"
        >
          Email address
        </label>
        <input
          autocomplete="email"
          class="w-72 rounded-md px-4 focus:border-transparent focus:outline-none focus:ring-2 focus:ring-primary-600 dark:bg-black"
          id="email-input"
          name="email"
          placeholder="Enter your email"
          required=""
          type="email"
        />
      </div>
      <div
        class="mt-2 flex w-full rounded-md shadow-sm sm:mt-0 sm:ml-3"
      >
        <button
          class="w-full rounded-md bg-primary-500 py-2 px-4 font-medium text-white sm:py-0 hover:bg-primary-700 dark:hover:bg-primary-400 focus:outline-none focus:ring-2 focus:ring-primary-600 focus:ring-offset-2 dark:ring-offset-black"
          type="submit"
        >
          Sign up
        </button>
      </div>
    </form>
  </div>
</div>
`;
